### PR TITLE
Made /update async and able to download next song

### PR DESCRIPTION
**Pull request by Taoshix**
## What is this fixing or adding?
It is changing /update to be async and download next song if auto download is true
It essentially just invoking a slightly modified version of the code used for game watcher, please see for yourself

## How was this tested?
Through blood sweat and tears with lots of trial and error on a local AP server instance
I should have read the existing code and just invoked that instead of trying (and failing) to recreate what already exists, but you live and learn

**Please test this too to make sure it works properly as it felt like I changed alot**

## If this makes graphical changes, please attach screenshots.
image

### CODE REVIEW

File: worlds/osu/Client.py [hunk 1]
```diff
@@ -558,6 +559,41 @@ async def auto_get_last_scores(ctx, mode=''):
             ctx.last_scores.pop(0)
         check_location(ctx, score)
 
+# This is the non-silent version of the function above where this one is used in the update command
+async def get_last_scores(self, mode=''):
+    # Make URl for the request
+    try:
+        request = f"https://osu.ppy.sh/api/v2/users/{os.environ['PLAYER_ID']}/scores/recent?include_fails=1&limit=100"
+    except KeyError:
+        self.output('No Player ID')
```
Comment (lilymnky-F) [inline worlds/osu/Client.py]: Change to "Set a Player ID" or similar to match similar errors

File: worlds/osu/Client.py [hunk 2]
```diff
@@ -558,6 +559,41 @@ async def auto_get_last_scores(ctx, mode=''):
...
+        return
+    # Add Mode to request, otherwise it will use the user's default
+    if mode:
```
Comment (lilymnky-F) [inline worlds/osu/Client.py]: Doesn't check if the gamemode is valid. Current Update Command uses:
`if mode and mode.lower() in self.mode_names.keys():`
`  request += f"&mode={self.mode_names[mode.lower()]}"`

File: worlds/osu/Client.py [hunk 3]
```diff
@@ -520,7 +487,41 @@ async def download_next_beatmapset_silent(ctx, task):
     except Exception as e:
         print(f"An error occurred: {e}")
 
+# This is the non-silent version of the function above where this one is used in the update command
+async def download_next_beatmapset(self, task):
```
Comment (lilymnky-F) [inline worlds/osu/Client.py:491]: having this be its own function is kinda akward but you're already planning a refactor so its fine

**Review by lilymnky-F — CHANGES_REQUESTED** [review]
Mode Checking is needed for update.

File: worlds/osu/Client.py (hunk 3 above)
Comment (Taoshix) [reply worlds/osu/Client.py:491]: I thought it would be out of scope for this PR to fix it, next one will fix it

**Pull request merged**